kubernetes/nodecollection.go

```diff
@@ -81,6 +81,10 @@ func downloadNodeData(prefix string,
 	}
 
 	for _, n := range nodes.Items {
+		if n.Spec.ProviderID == "" {
+			log.Println("Warning: Provider ID for node does not exist")
```

Comment:
Please also log the node name that does not have a providerID in the message as well as something along the lines of "If this condition persists it will cause inconsistent cluster allocation"

Reply:
Should we add a test for this?  Should be able to mock node Spec in a similar fashion to this:
https://github.com/cloudability/metrics-agent/blob/master/kubernetes/nodecollection_test.go#L24-L27